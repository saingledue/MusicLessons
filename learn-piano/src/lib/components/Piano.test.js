import React from 'react';
import { mount } from 'enzyme';
import Piano from './Piano';

function PianoKey({ note, startPlayingNote, stopPlayingNote }) {
  return <button data-note={note}
    onMouseDown={startPlayingNote}
    onMouseUp={stopPlayingNote}
  />;
}

function startPlayingPianoKey(wrapper, note) {
  wrapper.find(`button[data-note="${note}"]`).simulate('mousedown');
}

function stopPlayingPianoKey(wrapper, note) {
  wrapper.find(`button[data-note="${note}"]`).simulate('mouseup');
}

describe('Piano', () => {
  it('calls renderPianoKey() with the expected props', () => {
    const renderPianoKey = jest.fn(PianoKey);
    mount(<Piano startNote={'C1'}
      endNote={'C2'}
      renderPianoKey={renderPianoKey}
    />);

    expect(renderPianoKey.mock.calls).toMatchSnapshot();
  });

  it('calls renderAudio() with the expected props', () => {
    const renderAudio = jest.fn(() => null);
    const wrapper = mount(
      <Piano startNote={'C1'}
        endNote={'C2'}
        renderPianoKey={PianoKey}
        renderAudio={renderAudio}
      />
    );

    startPlayingPianoKey(wrapper, 'C1');
    startPlayingPianoKey(wrapper, 'C2');
    stopPlayingPianoKey(wrapper, 'C1');

    expect(renderAudio.mock.calls).toMatchSnapshot();
  });
});
