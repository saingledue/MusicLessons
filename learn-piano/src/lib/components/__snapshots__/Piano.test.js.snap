// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piano calls renderAudio() with the expected props 1`] = `
Array [
  Array [
    Object {
      "notes": Array [],
    },
    Object {},
  ],
  Array [
    Object {
      "notes": Array [
        "C1",
      ],
    },
    Object {},
  ],
  Array [
    Object {
      "notes": Array [
        "C1",
        "C2",
      ],
    },
    Object {},
  ],
  Array [
    Object {
      "notes": Array [
        "C2",
      ],
    },
    Object {},
  ],
]
`;

exports[`Piano calls renderPianoKey() with the expected props 1`] = `
Array [
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "D1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "D#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "E1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "F1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "F#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "G1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "G#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "A1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "A#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "B1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C2",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
]
`;
